### PR TITLE
Document S3 and network drives connectors

### PR DESCRIPTION
**Pull request by praveen-elastic**
#### Pre-Review Checklist
- [x] Covered the changes with automated tests
- [x] Tested the changes locally

#### Changes Requiring Extra Attention

- [x] Security-related changes (encryption, TLS, SSRF, etc)
- [x] New external service dependencies added.

### CODE REVIEW

**Review by tarekziade — COMMENTED** [review]
looks great! can we move the docs, one document per connector:

```
- docs/
-    connectors/
-                s3.md
-                network_drive.md
```

**Comment by danajuratoni** [comment]
As I commented on the issue, let's keep the current format and split later if needed.

File: docs/DEVELOPING.md [hunk 1]
```diff
@@ -1,5 +1,187 @@
 # Connectors Developer's Guide
 
+## Network Drives Connector
+
+Elastic Network Drives connector is provided in Elastic Connectors python framework and can be used via [build a connector](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html).
+
+## Availability and prerequisites
```
Comment (danajuratoni) [inline docs/DEVELOPING.md:7]: Please use Level 3 subheading for "Availability and prerequisites" (for both Network Drive and S3) similar to "Setup and basic usage"

File: docs/DEVELOPING.md [hunk 2]
```diff
@@ -1,5 +1,187 @@
 # Connectors Developer's Guide
 
+## Network Drives Connector
```
Comment (danajuratoni) [inline docs/DEVELOPING.md:3]: The connector is called Network **Drive**, no plural

**Comment by danajuratoni** [comment]
Also, information requested by Pius https://github.com/elastic/connectors-python/issues/260#issuecomment-1371330065 still needs to be added.

File: docs/DEVELOPING.md [hunk 3]
```diff
@@ -1,5 +1,187 @@
...
+
+Elastic Network Drives connector is provided in Elastic Connectors python framework and can be used via [build a connector](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html).
```
Comment (danajuratoni) [inline docs/DEVELOPING.md:5]: Let's link "Network Drives connector" above to /connectors/sources/network_drive.py
and similarly for S3 to s3.py.

File: docs/DEVELOPING.md [hunk 4]
```diff
@@ -1,5 +1,187 @@
...
+
+ℹ️ The drive path should have forward slashes as path separators.
+
+##### `enable_content_extraction`
+
+Whether the connector should extract file content from network drives. Default value is `True` i.e. the connector will try to extract file contents.
+
+#### Content extraction
+
+The connector uses Elastic ingest attachment processor plugin for extracting file contents. The ingest attachment processor extracts files by using Apache text extraction library Tika. Supported file types eligible for extraction can be found [here](https://github.com/elastic/connectors-python/blob/9cf07a96288dcd542b641c51533ee2d427ef56ae/connectors/utils.py#L27).
```
[the remark this answers is not in the document]
Comment (praveen-elastic) [reply docs/DEVELOPING.md:75]: The documentation https://www.elastic.co/guide/en/elasticsearch/reference/current/attachment.html does not mention the supported file types. In order to make it more clear to the user, we thought of adding the reference to the complete list of supported types here.

[the remark this answers is not in the document]
Comment (praveen-elastic) [reply docs/DEVELOPING.md:75]: As mentioned [here](https://github.com/elastic/connectors-python/pull/214#discussion_r1035191425) by @seanstory this list of supported types in our repo is the one Attachment Processor supports

**Comment by praveen-elastic** [comment]
> Also, information requested by Pius [#260 (comment)](https://github.com/elastic/connectors-python/issues/260#issuecomment-1371330065) still needs to be added.

I've added the limitation regarding s3 compatible vendors as mentioned here
https://github.com/elastic/connectors-python/pull/266#issuecomment-1377055886

File: docs/DEVELOPING.md (hunk 3 above)
Comment (danajuratoni) [reply docs/DEVELOPING.md]: I see the link goes to main. We should make sure the link references the 8.6 source when it's backported to 8.6.

File: docs/DEVELOPING.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (danajuratoni) [reply docs/DEVELOPING.md:75]: I see, in this case it makes sense to use the defined list 👍

File: docs/DEVELOPING.md [hunk 5]
```diff
@@ -1,5 +1,187 @@
...
+
+- The network drive path the connector will crawl to fetch files. This is the name of the folder shared via SMB.
+- The username the connector will use to log in to network drive.
+- The password the connector will use to log in to network drive.
+- The server IP address where the network drive is hosted.
+- The port where the network drive service is hosted.
+
+#### Configure Network Drive connector
+
+Following configuration fields need to be provided for setting up the connector:
```
Comment (chriscressman) [inline docs/DEVELOPING.md:34]:
```suggestion
The following configuration fields need to be provided for setting up the connector:
```

File: docs/DEVELOPING.md [hunk 6]
```diff
@@ -1,5 +1,187 @@
...
+
+ℹ️ The drive path should have forward slashes as path separators.
+
+##### `enable_content_extraction`
+
+Whether the connector should extract file content from network drive. Default value is `True` i.e. the connector will try to extract file contents.
+
+#### Content extraction
+
+The connector uses Elastic ingest attachment processor plugin for extracting file contents. The ingest attachment processor extracts files by using Apache text extraction library Tika. Supported file types eligible for extraction can be found [here](https://github.com/elastic/connectors-python/blob/9cf07a96288dcd542b641c51533ee2d427ef56ae/connectors/utils.py#L27).
```
Comment (chriscressman) [inline docs/DEVELOPING.md:75]:
```suggestion
The connector uses the Elastic ingest attachment processor plugin for extracting file contents. The ingest attachment processor extracts files by using the Apache text extraction library Tika. Supported file types eligible for extraction can be found [here](https://github.com/elastic/connectors-python/blob/9cf07a96288dcd542b641c51533ee2d427ef56ae/connectors/utils.py#L27).
```

File: docs/DEVELOPING.md [hunk 7]
```diff
@@ -1,5 +1,187 @@
...
+```shell
+$ make ftest NAME=network_drive
+```
+
+ℹ️ The e2e test uses default values defined in [configure Network Drive connector](#configure-network-drive-connector)
+
+
+## Amazon S3 Connector
+
+[Elastic Amazon S3 connector](https://github.com/elastic/connectors-python/blob/main/connectors/sources/s3.py) is used to sync files and file content for supported file types from [Amazon S3](https://s3.console.aws.amazon.com/s3/home) data sources. It is provided in Elastic Connectors python framework and can be used via [build a connector](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html).
```
Comment (chriscressman) [inline docs/DEVELOPING.md:95]:
```suggestion
The [Elastic Amazon S3 connector](https://github.com/elastic/connectors-python/blob/main/connectors/sources/s3.py) is used to sync files and file content for supported file types from [Amazon S3](https://s3.console.aws.amazon.com/s3/home) data sources. It is provided in the Elastic connectors python framework and can be used via [build a connector](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html).
```

File: docs/DEVELOPING.md [hunk 8]
```diff
@@ -1,5 +1,187 @@
...
+
+Collect the information that is required to connect to your amazon s3:
+
+- Setup aws configuration by installing [awscli](https://pypi.org/project/awscli/).
+- Add aws_access_key, aws_secret_key and region to run the connector.
+
+
+#### Configure Amazon S3 connector
+
+Following configuration fields need to be provided for setting up the connector:
```
Comment (chriscressman) [inline docs/DEVELOPING.md:121]:
```suggestion
The following configuration fields need to be provided for setting up the connector:
```

File: docs/DEVELOPING.md [hunk 9]
```diff
@@ -1,5 +1,187 @@
...
+  - `50`
+  - `150`
+
+##### `enable_content_extraction`
+
+Whether the connector should extract file content from amazon s3. Default value is `True` i.e. the connector will try to extract file contents.
+
+#### Content extraction
+
+The connector uses Elastic ingest attachment processor plugin for extracting file contents. The ingest attachment processor extracts files by using Apache text extraction library Tika. Supported file types eligible for extraction can be found [here](https://github.com/elastic/connectors-python/blob/9cf07a96288dcd542b641c51533ee2d427ef56ae/connectors/utils.py#L27).
```
Comment (chriscressman) [inline docs/DEVELOPING.md:164]:
```suggestion
The connector uses the Elastic ingest attachment processor plugin for extracting file contents. The ingest attachment processor extracts files by using the Apache text extraction library Tika. Supported file types eligible for extraction can be found [here](https://github.com/elastic/connectors-python/blob/9cf07a96288dcd542b641c51533ee2d427ef56ae/connectors/utils.py#L27).
```

File: docs/DEVELOPING.md [hunk 10]
```diff
@@ -1,5 +1,187 @@
...
+
+### Connector Limitations
+
+- Files with size greater than 10 MB won't be extracted.
+- Permission are not synced. All the documents indexed to an Elastic deployment will be visible to all the users having access to that Elastic Deployment.
+- Filtering rules are not available in the present version. Currently, the filtering is controlled via ingest pipelines.
+- User needs to set a profile with aws configure command.
```
Comment (chriscressman) [inline docs/DEVELOPING.md:171]:
```suggestion
- The user needs to set a profile with the aws configure command.
```

**Review by chriscressman — COMMENTED** [review]
I added a few minor suggestions, but this otherwise looks good to me.

File: docs/DEVELOPING.md [hunk 11]
```diff
@@ -1,5 +1,187 @@
...
+- Add aws_access_key, aws_secret_key and region to run the connector.
+
+
+#### Configure Amazon S3 connector
+
+Following configuration fields need to be provided for setting up the connector:
+
+##### `buckets`
+
+List buckets for amazon s3. For empty list connector will fetch data for all the buckets. Examples:
```
Comment (leemthompo) [inline docs/DEVELOPING.md:125]:
```suggestion
List buckets for amazon s3. If empty, the connector will fetch data for all the buckets. Examples:
```

File: docs/DEVELOPING.md [hunk 12]
```diff
@@ -1,5 +1,187 @@
...
+
+  - `[testbucket, prodbucket]`
+  - `[]`
+
+##### `read_timeout`
+
+The read_timeout for amazon s3. Default value is `90`. Examples:
```
Comment (leemthompo) [inline docs/DEVELOPING.md:132]:
```suggestion
The read_timeout for Amazon S3. Default value is `90`. Examples:
```

Be sure to use Amazon S3 (in capitals) throughout.

File: docs/DEVELOPING.md [hunk 13]
```diff
@@ -1,5 +1,187 @@
...
+Whether the connector should extract file content from amazon s3. Default value is `True` i.e. the connector will try to extract file contents.
+
+#### Content extraction
+
+The connector uses Elastic ingest attachment processor plugin for extracting file contents. The ingest attachment processor extracts files by using Apache text extraction library Tika. Supported file types eligible for extraction can be found [here](https://github.com/elastic/connectors-python/blob/9cf07a96288dcd542b641c51533ee2d427ef56ae/connectors/utils.py#L27).
+
+### Connector Limitations
+
+- Files with size greater than 10 MB won't be extracted.
+- Permission are not synced. All the documents indexed to an Elastic deployment will be visible to all the users having access to that Elastic Deployment.
```
Comment (leemthompo) [inline docs/DEVELOPING.md:169]:
```suggestion
- Permission are not synced. *All documents* indexed to an Elastic deployment will be visible to all the users having access to that Elastic Deployment.
```

File: docs/DEVELOPING.md (hunk 12 above)
Comment (leemthompo) [reply docs/DEVELOPING.md:132]: Similarly use AWS not aws

File: docs/DEVELOPING.md [hunk 14]
```diff
@@ -1,5 +1,187 @@
...
+- Filtering rules are not available in the present version. Currently, the filtering is controlled via ingest pipelines.
+- User needs to set a profile with aws configure command.
+- Current version of the connector does not support S3 compatible vendors.
```
Comment (leemthompo) [inline docs/DEVELOPING.md:172]:
```suggestion
- Currently the connector does not support S3 compatible vendors.
```

File: docs/DEVELOPING.md [hunk 15]
```diff
@@ -1,5 +1,187 @@
...
+
+### E2E Tests
+
+The framework allows users to test the connector end to end. To perform e2e test for Amazon S3 connector, run the following make command:
```
Comment (leemthompo) [inline docs/DEVELOPING.md:176]:
```suggestion
The framework allows users to test the connector end to end (e2e). To perform e2e tests for the Amazon S3 connector, run the following `make` command:
```

File: docs/DEVELOPING.md (hunk 3 above)
Comment (praveen-elastic) [reply docs/DEVELOPING.md]: > I see the link goes to main. We should make sure the link references the 8.6 source when it's backported to 8.6.

The code in 8.6 branch in not updated. Previously for s3 we had aws.py file but now it is renamed to s3.py. So as of now is it fine if we can use the main branch link? Once the code is backported to 8.6 we'll update the document to use the 8.6 branch.

**Comment by praveen-elastic** [comment]
@danajuratoni @leemthompo @chriscressman I've updated the PR with your respective suggestions. Please re-review it.

**Review by leemthompo — APPROVED** [review]
LGTM 👍

File: docs/DEVELOPING.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (seanstory) [reply docs/DEVELOPING.md]: To clarify, the Attachment Processor supports a much wider range of files. However, I suggested that, as a starting point, we claim support for the file types that we support in Workplace Search, and we add from there. This keeps us from saying "yeah, we support X type" when we actually don't know anything about the quality of output you will get for X. And there are MANY file types that tika technically "supports" where the output is non-ideal by default.

File: docs/DEVELOPING.md (hunk 3 above)
Comment (danajuratoni) [reply docs/DEVELOPING.md]: That's fine, as long you make sure the 8.6 branch has the correct links.

**Comment by danajuratoni** [comment]
Looks good!

**Review by danajuratoni — APPROVED** [review]

**Review by wangch079 — APPROVED** [review]

**Comment by vidok** [comment]
buildkite test this

**Pull request merged**